File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(),strs.end());
        int cnt=0,n=strs.size();
        for(int i=0;i<strs[0].size();i++){
            if(strs[0][i] == strs[n-1][i])
            cnt++;
            else
            break;
        }
        return strs[0].substr(0,cnt);
    }
};

File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);
        int cnt=0,n=strs.length;
        for(int i=0;i<Math.min(strs[0].length() , strs[n-1].length());i++){
            if(strs[0].charAt(i) != strs[n-1].charAt(i))
            break;
            cnt++;
        }
        return strs[0].substring(0,cnt);
    }
}